mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java

```diff
@@ -22,16 +22,123 @@
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Arrays;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
 
 public class Attachment {
 
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static Attachment from(byte[] bytes, String type) {
+        return builder()
+                .attachmentId(AttachmentId.forPayload(bytes))
+                .bytes(bytes)
+                .type(type)
+                .size(bytes.length)
+                .build();
+    }
+
+    public static class Builder {
+
+        private AttachmentId attachmentId;
+        private byte[] bytes;
+        private String type;
+        private Long size;
+
+        private Builder() {
+        }
+
+        public Builder attachmentId(AttachmentId attachmentId) {
+            Preconditions.checkArgument(attachmentId != null);
+            this.attachmentId = attachmentId;
+            return this;
+        }
+
+        public Builder bytes(byte[] bytes) {
+            Preconditions.checkArgument(bytes != null);
+            this.bytes = bytes;
+            return this;
+        }
+
+        public Builder type(String type) {
+            Preconditions.checkArgument(!Strings.isNullOrEmpty(type));
+            this.type = type;
+            return this;
+        }
+
+        public Builder size(long size) {
+            this.size = size;
+            return this;
+        }
+
+        public Attachment build() {
+            Preconditions.checkState(attachmentId != null, "'attachmentId' is mandatory");
+            Preconditions.checkState(bytes != null, "'bytes' is mandatory");
+            Preconditions.checkState(type != null, "'type' is mandatory");
+            Preconditions.checkState(size != null, "'size' is mandatory");
```

Comment:
As size is just an information, and don't know the benefit of such check.
It may break, if the payload returned by the backend doesn't have the same size as the one store in this backend.